### ISSUE TITLE
[Feature Request] Block templates can be overridden like Page model 

### ISSUE THREAD

**Issue by cbledsoe** [wagtail/wagtail#3857]
Currently the Wagtail page model allows you to override the template on instances like so:
http://docs.wagtail.io/en/latest/topics/pages.html?highlight=get_template#dynamically-choosing-the-template

I would like to propose a change to allow the same methodology on Block classes like so:

```
class SomeBlock(StructBlock):

    def get_template(self):
        if self.use_other_template:
            return 'blocks/alternate_some_block.html'

        return 'blocks/some_block.html'
```

I will be submitting a Pull Request to attach to this issue with a proposed implementation.

#3858

**Comment by tkwon** [comment]
The answer might be in this explanation:

https://github.com/wagtail/wagtail/issues/4387

and here:

http://docs.wagtail.io/en/v2.0/topics/streamfield.html#custom-value-class-for-structblock

**Comment by gasman** [comment]
Implemented in #3858
